batch2/create-batch-tables.sql

```diff
@@ -195,6 +204,10 @@ BEGIN
 
   SELECT state INTO cur_state FROM instances WHERE name = in_instance_name;
 
+  UPDATE instances
+  SET time_deactivated = in_timestamp
+  WHERE name = in_instance_name;
```

Comment:
I think you should do this in the same statement below where the state is set to `inactive`.

Reply:
I think it needs to be here. Otherwise, we're only getting the first deactivation time and there could be earlier times that are called later. That's why the attempts table update is outside the if clause as well.